### PR TITLE
Remove unused macros

### PR DESCRIPTION
**Pull request by raveit65**

### CODE REVIEW

**Review by lukefromdc — APPROVED** [review]
I just got a successful build and install, no runtime warnings in a terminal from which `sudo caja` was run. Could not tell if there were any NEW build warnings due to the usual deprecation warning spew but the total number of build warnings is definately trending down as we work on them, lots of files building w no warnings now

File: src/file-manager/fm-icon-container.c [hunk 1]
```diff
@@ -36,8 +36,6 @@
 
 #include "fm-icon-container.h"
 
-#define ICON_TEXT_ATTRIBUTES_NUM_ITEMS		3
-#define ICON_TEXT_ATTRIBUTES_DEFAULT_TOKENS	"size,date_modified,type"
```
Comment (rbuj) [inline src/file-manager/fm-icon-container.c:39]: avoid adding two consecutive blank lines

File: src/caja-trash-bar.c [hunk 2]
```diff
@@ -33,8 +33,6 @@
 #include "caja-trash-bar.h"
 #include "caja-window.h"
 
-#define CAJA_TRASH_BAR_GET_PRIVATE(o)\
-	(G_TYPE_INSTANCE_GET_PRIVATE ((o), CAJA_TYPE_TRASH_BAR, CajaTrashBarPrivate))
```
Comment (rbuj) [inline src/caja-trash-bar.c:36]: avoid adding two consecutive blank lines

File: eel/eel-gdk-extensions.c [hunk 3]
```diff
@@ -37,7 +37,6 @@
 #include <stdlib.h>
 #include <pango/pango.h>
 
-#define GRADIENT_BAND_SIZE 4
```
Comment (rbuj) [inline eel/eel-gdk-extensions.c:40]: avoid adding two consecutive blank lines

File: eel/eel-stock-dialogs.c [hunk 4]
```diff
@@ -36,7 +36,6 @@
 
 #define TIMED_WAIT_MINIMUM_DIALOG_WIDTH 300
 
-#define RESPONSE_DETAILS 1000
```
Comment (rbuj) [inline eel/eel-stock-dialogs.c:39]: avoid adding two consecutive blank lines

File: src/caja-history-sidebar.c [hunk 5]
```diff
@@ -43,12 +43,6 @@
 
 #include "caja-history-sidebar.h"
 
-#define CAJA_HISTORY_SIDEBAR_CLASS(klass) \
-  (G_TYPE_CHECK_CLASS_CAST ((klass), CAJA_TYPE_HISTORY_SIDEBAR, CajaHistorySidebarClass))
-#define CAJA_IS_HISTORY_SIDEBAR(obj) \
-  (G_TYPE_CHECK_INSTANCE_TYPE ((obj), CAJA_TYPE_HISTORY_SIDEBAR))
-#define CAJA_IS_HISTORY_SIDEBAR_CLASS(klass) \
-  (G_TYPE_CHECK_CLASS_TYPE ((klass), CAJA_TYPE_HISTORY_SIDEBAR))
```
Comment (rbuj) [inline src/caja-history-sidebar.c:46]: avoid adding two consecutive blank lines

File: src/caja-places-sidebar.c [hunk 6]
```diff
@@ -58,7 +58,7 @@
 #include "caja-window.h"
 
 #define EJECT_BUTTON_XPAD 6
-#define ICON_CELL_XPAD 6
+
```
Comment (rbuj) [inline src/caja-places-sidebar.c:62]: avoid adding two consecutive blank lines

**Comment by raveit65** [comment]
@rbuj 
Done, PR is updated.

File: libcaja-private/caja-file.c [hunk 7]
```diff
@@ -87,17 +84,12 @@
 #ifdef CAJA_FILE_DEBUG_REF_VALGRIND
 #include <valgrind/valgrind.h>
 #define DEBUG_REF_PRINTF VALGRIND_PRINTF_BACKTRACE
-#else
-#define DEBUG_REF_PRINTF printf
```
Comment (cwendling) [inline libcaja-private/caja-file.c:91]: This is not correct IMO.  `DEBUG_REF_PRINTF` is used 4 times, and it's guarded by `CAJA_FILE_DEBUG_REF` *not* `CAJA_FILE_DEBUG_REF_VALGRIND`.  So although bot `CAJA_FILE_DEBUG_REF` and `CAJA_FILE_DEBUG_REF_VALGRIND` are undef'd on lines 84 and 85 respectively, I would think this serves as debugging switches -- make one or both `#define` and you get the debugging behavior.

In any case, removing this breaks the build with `CAJA_FILE_DEBUG_REF` defined and `CAJA_FILE_DEBUG_REF_VALGRIND` *not* defined.

File: libcaja-private/caja-module.c [hunk 8]
```diff
@@ -32,9 +32,6 @@
 
 #define CAJA_TYPE_MODULE    	(caja_module_get_type ())
 #define CAJA_MODULE(obj)		(G_TYPE_CHECK_INSTANCE_CAST ((obj), CAJA_TYPE_MODULE, CajaModule))
-#define CAJA_MODULE_CLASS(klass)	(G_TYPE_CHECK_CLASS_CAST ((klass), CAJA_TYPE_MODULE, CajaModule))
-#define CAJA_IS_MODULE(obj)		(G_TYPE_INSTANCE_CHECK_TYPE ((obj), CAJA_TYPE_MODULE))
-#define CAJA_IS_MODULE_CLASS(klass)	(G_TYPE_CLASS_CHECK_CLASS_TYPE ((klass), CAJA_TYPE_MODULE))
```
Comment (cwendling) [inline libcaja-private/caja-module.c:37]: These are standard defines for a GType, but as they are not needed and private, OK.

File: src/caja-bookmarks-sidebar.c [hunk 9]
```diff
@@ -50,12 +50,6 @@
 
 #include "caja-bookmarks-sidebar.h"
 
-#define CAJA_BOOKMARKS_SIDEBAR_CLASS(klass) \
-  (G_TYPE_CHECK_CLASS_CAST ((klass), CAJA_TYPE_BOOKMARKS_SIDEBAR, CajaBookmarksSidebarClass))
-#define CAJA_IS_BOOKMARKS_SIDEBAR(obj) \
-  (G_TYPE_CHECK_INSTANCE_TYPE ((obj), CAJA_TYPE_BOOKMARKS_SIDEBAR))
-#define CAJA_IS_BOOKMARKS_SIDEBAR_CLASS(klass) \
-  (G_TYPE_CHECK_CLASS_TYPE ((klass), CAJA_TYPE_BOOKMARKS_SIDEBAR))
```
Comment (cwendling) [inline src/caja-bookmarks-sidebar.c:58]: Here too, standard but unused.

File: src/caja-history-sidebar.c [hunk 10]
```diff
@@ -43,13 +43,6 @@
...
-
```
Comment (cwendling) [inline src/caja-history-sidebar.c:52]: Ditto.

**Review by cwendling — CHANGES_REQUESTED** [review]
Mostly good, but I think the `DEBUG_REF_PRINTF` is wrong.  The rest looks OK.

File: libcaja-private/caja-file.c (hunk 7 above)
Comment (raveit65) [reply libcaja-private/caja-file.c]: Done, i added the define back.

File: libcaja-private/caja-module.c (hunk 8 above)
Comment (raveit65) [reply libcaja-private/caja-module.c:37]: Should i add them back?

File: src/caja-bookmarks-sidebar.c (hunk 9 above)
Comment (raveit65) [reply src/caja-bookmarks-sidebar.c:58]: Should i add them back?
Or is it OK?

File: src/caja-history-sidebar.c (hunk 10 above)
Comment (raveit65) [reply src/caja-history-sidebar.c:52]: OK?

File: libcaja-private/caja-file.c (hunk 7 above)
Comment (cwendling) [reply libcaja-private/caja-file.c]: Great :+1: 

Alternatively if the unused macro warning really needs to go, you could actually replace the `#else` with `#elif defined(CAJA_FILE_DEBUG_REF)`, and it should be all good as all use are guarded with `CAJA_FILE_DEBUG_REF`.
I'm happy with the current status, but either way works for me.

**Review by cwendling — APPROVED** [review]

File: libcaja-private/caja-module.c (hunk 8 above)
Comment (raveit65) [reply libcaja-private/caja-module.c:37]: Ahh, it is OK . I over read your first comment :)

Comment (cwendling) [reply libcaja-private/caja-module.c:37]: No need if we're all happy not to have them.  I just wanted to mention that those might actually be useful at some point, and are not mere letfover of days long past; but as they don't appear in a header, it's OK to remove them if we want.

**Comment by raveit65** [comment]
Thanks to @sc0w

**Pull request merged**